app/controllers/tasks.js

```diff
@@ -0,0 +1,92 @@
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { action, set } from '@ember/object';
+import ENV from 'website-my/config/environment';
+
+const API_BASE_URL = ENV.BASE_API_URL;
+
+export default class TasksController extends Controller {
+  @tracked showDropDown = false;
+  @tracked showActiveTasks = true;
+  @tracked showPendingTasks = false;
+  @tracked showBlockedTasks = false;
+  @tracked activeTasksList = this.model.activeTasks;
+  @tracked blockedTasksList = this.model.blockedTasks;
+  @tracked pendingTasksList = this.model.pendingTasks;
+  @tracked allTasksList = this.model.tasks;
+
+  @tracked fieldsToBeUpdated = {};
+  @tracked collection = null;
+
+  @action toggleDropDown() {
+    this.showDropDown = !this.showDropDown;
+  }
+
+  @action toggleActiveTasks() {
+    this.showActiveTasks = true;
+    this.showBlockedTasks = false;
+    this.showPendingTasks = false;
+  }
+
+  @action togglePendingTasks() {
+    this.showActiveTasks = false;
+    this.showBlockedTasks = false;
+    this.showPendingTasks = true;
+  }
+
+  @action toggleBlockedTasks() {
+    this.showActiveTasks = false;
+    this.showBlockedTasks = true;
+    this.showPendingTasks = false;
+  }
+
+  @action onTaskChange(key, value) {
+    this.fieldsToBeUpdated[key] = value;
```

Comment:
What do these field denote? What are they holding? Can we get a bit more specific please?

Reply:
Updated